### PR TITLE
Make templates support Azure Cognitive Search

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
We want the deployment templates support ACS

### Description
Augment the templates so that they support ACS in addition to Qdrant for storage of embeddings

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md#dev-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: deploy/deploy-azure.ps1 [hunk 1]
```diff
@@ -34,15 +34,16 @@ param(
 
     [string]
     # Region to which to make the deployment (ignored when deploying to an existing resource group)
-    $Region = "centralus",
+    $Region = "southcentralus",
```
Comment (glahaye) [inline deploy/deploy-azure.ps1:37]: Central US no longer has quota capacity

File: deploy/deploy-webapp.ps1 [hunk 2]
```diff
@@ -94,8 +94,10 @@ if ($LASTEXITCODE -ne 0) {
     exit $LASTEXITCODE
 }
 
+Pop-Location
```
Comment (glahaye) [inline deploy/deploy-webapp.ps1:97]: Revert dir sooner, in case we fail down the line

**Review by dehoward — DISMISSED** [review]

File: deploy/deploy-azure.sh [hunk 3]
```diff
@@ -17,7 +17,7 @@ usage() {
     echo "  -rg, --resource-group RESOURCE_GROUP       Resource group to which to make the deployment (default: \"rg-\$DEPLOYMENT_NAME\")"
     echo "  -r, --region REGION                        Region to which to make the deployment (default: \"South Central US\")"
     echo "  -a, --app-service-sku WEB_APP_SVC_SKU      SKU for the Azure App Service plan (default: \"B1\")"
-    echo "  -nq, --no-qdrant                           Don't deploy Qdrant for memory storage - Use volatile memory instead"
+    echo "  -ms, --memory-store                        Method to use to persist embeddings (default: \"AzureCognitiveSearch\")"
```
Comment (alliscode) [inline deploy/deploy-azure.sh:20]: We should list the valid options here: ""Volatile", "AzureCognitiveSearch", "Qdrant""

**Review by alliscode — APPROVED** [review]

**Review by TaoChenOSU — DISMISSED** [review]

File: deploy/deploy-azure.sh (hunk 3 above)
Comment (glahaye) [reply deploy/deploy-azure.sh]: Issue addressed

**Review by alliscode — APPROVED** [review]

**Pull request merged**